jax/_src/numpy/fft.py

```diff
@@ -55,7 +58,20 @@ def _fft_core(func_name, fft_type, a, s, axes, norm):
     axes = tuple(range(a.ndim - len(axes), a.ndim))
     a = jnp.moveaxis(a, orig_axes, axes)
 
-  if s is None:
+  if s is not None:
+    a = jnp.asarray(a)
+    in_s = list(a.shape)
+    for axis, x in zip(axes, s):
+      in_s[axis] = x
+    if fft_type == xla_client.FftType.IRFFT:
+      in_s[-1] = (in_s[-1] // 2 + 1)
```

Comment:
Does this correctly handle the where the FFT is not along the last axis?

Perhaps this should be something like:
```suggestion
      in_s[axes[-1]] = (in_s[axes[-1]] // 2 + 1)
```

Reply:
This ensures that the last axis of the fft is also the last axis of `a`:
https://github.com/google/jax/blob/d5154a43c41dc5f6a56f2acf2ca590b4fdd873ca/jax/_src/numpy/fft.py#L56-L59